visited_fields_recorder.go

```diff
@@ -0,0 +1,34 @@
+package ucfg
+
+// VisitedFieldsRecorder keeps track of the visited fields in the configuration.
+type VisitedFieldsRecorder struct {
+	fields map[string]bool
+}
```

Comment:
change `fields` to `fields map[string]struct{}`. Then the value itself no by of 0 size.

Add a field to the set: `fields[name] = struct{}`.

Checking a field exists: `_, exists := fields[name]` ...

Reply:
Didn't know that.